activiti-cloud-modeling-service/activiti-cloud-services-modeling/activiti-cloud-services-modeling-service/src/main/java/org/activiti/cloud/services/modeling/validation/JsonSchemaModelValidator.java

```diff
@@ -41,111 +43,145 @@
-        } catch (ValidationException validationException) {
-            log.debug("Semantic model validation errors encountered: " + validationException.toJSON(),
-                      validationException);
-            throw new SemanticModelValidationException(validationException.getMessage(),
-                                                       getValidationErrors(validationException, processExtensionJson));
-        }
+  private final Logger log = LoggerFactory.getLogger(JsonSchemaModelValidator.class);
+
+  protected abstract SchemaLoader schemaLoader();
+
+  @Override
+  public void validate(byte[] modelContent,
+                       ValidationContext validationContext) {
+    JSONObject processExtensionJson = null;
+    try {
+      log.debug("Validating json model content: " + new String(modelContent));
+      processExtensionJson = extractJsonObject(modelContent);
+      validateJson(processExtensionJson);
+    } catch (JSONException jsonException) {
+      log.debug("Syntactic model JSON validation errors encountered",
+                jsonException);
+      throw new SyntacticModelValidationException(jsonException);
+    } catch (ValidationException validationException) {
+      log.debug("Semantic model validation errors encountered: " + validationException.toJSON(),
+                validationException);
+      throw new SemanticModelValidationException(validationException.getMessage(),
+                                                 getValidationErrors(validationException, processExtensionJson));
     }
-
-    private List<ModelValidationError> getValidationErrors(ValidationException validationException, JSONObject prcessExtenstionJson) {
-        return getValidationExceptions(validationException)
-                .map(exception -> toModelValidationError(exception, prcessExtenstionJson))
-                .distinct()
-                .collect(Collectors.toList());
+  }
+
+  @Override
+  public Collection<ModelValidationError> validateAndReturnErrors(byte[] modelContent,
+                                                                  ValidationContext validationContext) {
+    JSONObject processExtensionJson = null;
+    String jsonContent = new String(modelContent);
+    try {
+      log.debug("Validating json model content: " + jsonContent);
+      processExtensionJson = extractJsonObject(modelContent);
+      validateJson(processExtensionJson);
+    } catch (JSONException jsonException) {
+      log.debug("Syntactic model JSON validation errors encountered",
+                jsonException);
+      String problem = "Could not parse:" + jsonContent;
+      return List.of(new ModelValidationError(problem, jsonException.getMessage()));
+    } catch (ValidationException validationException) {
+      log.debug("Semantic model validation errors encountered: " + validationException.toJSON(),
+                validationException);
+      return getValidationErrors(validationException, processExtensionJson);
```

Comment:
I think we can try to reduce the code duplication between `validate` and `validateAndReturnErrors`, if we decide to keep both of them.

Reply:
Seen this problem so moved core logic into methods that are reused in both cases. 
Not sure if there is a room for more changes, because those cases having different signatures, anyway I'm open for discussion